test/test.js

```diff
@@ -35,6 +35,29 @@ test('queue concurrency', function (t) {
   t.equal(e.length, 1)
 })
 
+test('queue released when full', function (t) {
+  t.plan(20)
+
+  const e = mq({ concurrency: 1 })
+
+  e.on('hello 1', function (message, cb) {
+    t.ok(true, 'message received')
+    setTimeout(cb, 10)
+  })
+
+  function onSent () {
+    t.ok(true, 'message sent')
+  }
+
+  for (let i = 0; i < 9; i++) {
+    e._messageQueue.push({ topic: 'hello 1' })
+    e._messageCallbacks.push(onSent)
```

Comment:
Write a test without using private properties. I still do not understand what this is doing and my understanding of it is that it is not correct.

Reply:
This is related to: 
https://github.com/moscajs/aedes/issues/553
https://github.com/moscajs/aedes/issues/666

Reply:
I'm not able to reproduccec the error, BTW this is a safe approach to ensure the queue is always released when full

Reply:
That's the point of the limit. Once it is full, wait for one of the slots to be free.

This is equivalent to disabling the limit.

Reply:
> That's the point of the limit. Once it is full, wait for one of the slots to be free.

I know that and it's still working in that way, but for some reasons seems it could happen that the queue gets full and it's never released, for this reason I have added the `doing` flag that handles this specific case, if `doing` is true it means that the fastparallel is doing his job and will automatically call a release when done (this will ensure the concurrency is always respected as before), when doing is false and we add a message to the queue that message will never be processed as there is no parallel job active that once finished can process new messages and all the process will be stucked forever